tools/functional-tester/etcd-tester/lease_stresser.go

```diff
@@ -0,0 +1,373 @@
+	return func(mem *member, i int) LeaseStresser {
+		return &leaseStresser{
+			Endpoint:          mem.grpcAddr(),
+			numOfLeases:       m,
+			numOfKeysPerLease: n,
+			id:                i,
+		}
+	}
+}
+
+func (ls *leaseStresser) Setup() error {
+	if ls.numOfLeases == 0 {
+		panic("expect numOfLeases to be set")
+	}
+
+	if ls.numOfKeysPerLease == 0 {
+		panic("expect numOfKeysPerLease to be set")
+	}
+
+	conn, err := grpc.Dial(ls.Endpoint, grpc.WithInsecure())
+	if err != nil {
+		return fmt.Errorf("%v (%s)", err, ls.Endpoint)
+	}
+	ls.kvc = pb.NewKVClient(conn)
+	ls.lc = pb.NewLeaseClient(conn)
+	ls.conn = conn
+	ls.aliveLeases = make(map[int64]struct{})
+	ls.revokedLeases = make(map[int64]struct{})
+
+	return nil
+}
+
+func (ls *leaseStresser) Stress() error {
+	plog.Printf("lease Stresser %v starting ...", ls)
+	ctx, cancel := context.WithCancel(context.Background())
+	ls.cancel = cancel
+	ls.run(ctx)
+	return nil
+}
+
+func (ls *leaseStresser) run(ctx context.Context) {
+
+	ls.createLeases(ctx)
+	ls.randomlyDropLeases(ctx)
+
+}
+
+func (ls *leaseStresser) createLeases(ctx context.Context) {
+	numOfLeasesToCreate := ls.numOfLeases - len(ls.aliveLeases)
+	var wg sync.WaitGroup
+	wg.Add(numOfLeasesToCreate)
+	for i := 0; i < numOfLeasesToCreate; i++ {
+		go func() {
+			defer wg.Done()
+			leaseID, err := ls.createLeaseAndKeepAlive(ctx)
+
+			if err != nil {
+				return
+			}
+			plog.Printf("Lease %v created", leaseID)
+			ls.attachKeysWithLease(ctx, leaseID)
+			ls.mu.Lock()
+			ls.aliveLeases[leaseID] = struct{}{}
+			plog.Printf("Alive leases %v", ls.aliveLeases)
+			ls.mu.Unlock()
+
+		}()
+	}
+	wg.Wait()
+}
+
+func (ls *leaseStresser) randomlyDropLeases(ctx context.Context) {
+	var wg sync.WaitGroup
+	wg.Add(len(ls.aliveLeases))
+	for l := range ls.aliveLeases {
+		go func(leaseID int64) {
+			defer wg.Done()
+			isLeaseDropped := ls.randomlyDropLease(ctx, leaseID)
+			if isLeaseDropped {
+				ls.mu.Lock()
+				plog.Printf("Lease %v dropped", leaseID)
+				ls.revokedLeases[leaseID] = struct{}{}
+				delete(ls.aliveLeases, leaseID)
+				ls.mu.Unlock()
+			}
+		}(l)
+	}
+	wg.Wait()
+
+}
+
+func (ls *leaseStresser) InvariantCheck(ctx context.Context) error {
+	plog.Printf("Revoked Leases %+v ", ls.revokedLeases)
+	for leaseID := range ls.revokedLeases {
+		plog.Printf("Revoked Lease %v ", leaseID)
+		keysExpired, err := ls.hasKeysExpired(ctx, leaseID)
+		if err != nil {
+			return err
+		}
+
+		leaseExpired, err := ls.hasLeaseExpired(ctx, leaseID)
+		if err != nil {
+			return err
+		}
+
+		if !keysExpired || !leaseExpired {
+			return fmt.Errorf("revoked lease %v not expired", leaseID)
+		}
+	}
+
+	ls.mu.Lock()
+	ls.revokedLeases = make(map[int64]struct{})
+	ls.mu.Unlock()
+
+	// for those leases that are alive, checks whether if they still are after waiting for 2 * TTL
+	if len(ls.aliveLeases) != 0 {
+		time.Sleep(TTL * 2 * time.Second)
+	}
+
+	plog.Printf("Alive Leases %+v ", ls.aliveLeases)
```

Comment:
we shouldn't since I use lease keep alive on all alive  leases. I want to make sure even after sleep with 2 * TTL, the alive leases are still there. this check ensures that lease keep alive is also working.